src/gt4py/next/program_processors/runners/dace/gtir_sdfg.py

```diff
@@ -785,20 +793,36 @@ def visit_FunCall(
         elif cpm.is_applied_as_fieldop(node):
             return gtir_builtin_translators.translate_as_fieldop(node, sdfg, head_state, self)
         elif isinstance(node.fun, gtir.Lambda):
-            lambda_args = [
-                self.visit(
-                    arg,
-                    sdfg=sdfg,
-                    head_state=head_state,
+            # Special handling of lambda arguments that can be lowered as symbolic expressions:
+            # when all the symrefs the lambda argument depends on are SDFG symbols, the argument
+            # can be passed to the nested SDFG by means of symbol mapping.
+            symbolic_args = {}
+            for p, lambda_arg in zip(node.fun.params, node.args, strict=True):
+                if not isinstance(lambda_arg.type, ts.ScalarType):
+                    continue
+                symrefs = symbol_ref_utils.collect_symbol_refs(
+                    lambda_arg, self.global_symbols.keys()
                 )
-                for arg in node.args
-            ]
-
+                if all(symref in sdfg.symbols for symref in symrefs):
+                    symbolic_args[str(p.id)] = gtir_sdfg_utils.get_symbolic(lambda_arg)
+            # All other lambda arguments are lowered to taskgraphs that produce a data node.
+            args = {
+                str(p.id): (
```

Comment:
```suggestion
                param: (
```
Although I am not sure if this is possible, i.e. if `param` is still in scope.

Reply:
I tried that: it does not work.